test/src/code_field/tabs_to_spaces_test.dart

```diff
@@ -21,203 +20,218 @@ class Foo {
+        TextEditingValue value = TextEditingValue.empty.copyWith(
+          selection: const TextSelection.collapsed(offset: 0),
+        );
 
-      value = value.tabsToSpaces(_spaceCount);
+        value = value.tabsToSpaces(_spaceCount);
 
-      final tabsCount = RegExp('\t').allMatches(_codeWithTabs).length;
-      expect(value.text, _codeWithDoubleSpaces);
-      expect(value.selection.baseOffset, cursorPosition + tabsCount);
-      expect(value.selection.extentOffset, cursorPosition + tabsCount);
+        expect(value.text, '');
+        expect(value.selection, const TextSelection.collapsed(offset: 0));
+      });
     });
 
-    test('Code with tabs with selection', () {
-      const selectionStart = 24;
-      const selectionEnd = 59;
-      TextEditingValue value = const TextEditingValue(
-        text: _codeWithTabs,
-        selection: TextSelection(
-          baseOffset: selectionStart,
-          extentOffset: selectionEnd,
-        ),
-      );
-
-      value = value.tabsToSpaces(_spaceCount);
-
-      expect(value.text, _codeWithDoubleSpaces);
-      expect(value.selection.baseOffset, selectionStart + 1);
-      expect(value.selection.extentOffset, selectionEnd + 5);
+    group('Code without tabs', () {
+      test('without selection', () {
+        TextEditingValue value =
+            const TextEditingValue(text: _codeWithDoubleSpaces);
+
+        value = value.tabsToSpaces(_spaceCount);
+
+        expect(value.text, _codeWithDoubleSpaces);
+        expect(value.selection, const TextSelection.collapsed(offset: -1));
+      });
+
+      test('with cursor at the start', () {
+        TextEditingValue value = const TextEditingValue(
+          text: _codeWithDoubleSpaces,
+          selection: TextSelection.collapsed(offset: 0),
+        );
+
+        value = value.tabsToSpaces(_spaceCount);
+
+        expect(value.text, _codeWithDoubleSpaces);
+        expect(value.selection, const TextSelection.collapsed(offset: 0));
+      });
+
+      test('with cursor at the middle', () {
+        const cursorPosition = 5;
+        TextEditingValue value = const TextEditingValue(
+          text: _codeWithDoubleSpaces,
+          selection: TextSelection.collapsed(
+            offset: cursorPosition,
+            affinity: TextAffinity.upstream,
+          ),
+        );
+
+        value = value.tabsToSpaces(_spaceCount);
+
+        expect(value.text, _codeWithDoubleSpaces);
+        expect(
+          value.selection,
+          const TextSelection.collapsed(
+            offset: cursorPosition,
+            affinity: TextAffinity.upstream,
+          ),
+        );
+      });
+
+      test('with cursor at the end', () {
+        const cursorPosition = _codeWithDoubleSpaces.length;
+        TextEditingValue value = TextEditingValue(
+          text: _codeWithDoubleSpaces,
+          selection: const TextSelection.collapsed(offset: cursorPosition)
+              .copyWith(isDirectional: true),
```

Comment:
Per docs, a directional selection only makes sense for non-collapsed selection. That is why `collapsed` constructor does not have `isDirectional` argument.
Please test `isDirecional` in any of the non-collapsed selections below.